packages/common/src/feedback/_internal/computeProps.ts

```diff
@@ -0,0 +1,46 @@
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { UserSession } from "@esri/arcgis-rest-auth";
+import { getItemThumbnailUrl } from "../../resources";
+import { IModel } from "../../types";
+import { IHubFeedback } from "../../core";
+import { computeBaseProps } from "../../core/_internal/computeBaseProps";
+import { isDiscussable } from "../..";
+
+/**
+ * Given a model and a feedback object, set various computed properties that can't be directly mapped
+ * @private
+ * @param model
+ * @param feedback
+ * @param requestOptions
+ * @returns an IHubFeedback object
+ */
+export function computeProps(
+  model: IModel,
+  feedback: Partial<IHubFeedback>,
+  requestOptions: IRequestOptions
+): IHubFeedback {
+  let token: string;
+  if (requestOptions.authentication) {
+    const session: UserSession = requestOptions.authentication as UserSession;
+    token = session.token;
+  }
+  // compute base properties on feedback object
+  feedback = computeBaseProps(model.item, feedback);
+
+  // thumbnail url
+  feedback.thumbnailUrl = getItemThumbnailUrl(
+    model.item,
+    requestOptions,
+    token
+  );
+
+  // Handle Dates
+  feedback.createdDate = new Date(model.item.created);
+  feedback.createdDateSource = "item.created";
+  feedback.updatedDate = new Date(model.item.modified);
+  feedback.updatedDateSource = "item.modified";
+  feedback.isDiscussable = isDiscussable(feedback);
```

Comment:
We'll need to compute a couple custom properties here, though after thinking about this more, we won't actually need the `questions` on the entity. We should make a minor tweak/update to the [Hub Class Model](https://lucid.app/lucidchart/066ce98b-9aad-43d1-b7bd-36f5e89d0293/edit?invitationId=inv_1138f4b7-7396-4d29-ae18-06f287ce7eb7&page=E7lKAams_xlW#), specifically replace the `questions` member with the property enumerated in the second bullet point below.

1. `hasMapQuestion` - value is `true` when a map question exists for the survey, `false` when no map question exists for the survey (will control whether the `Map display` section render in the `settings` workspace pane, per [figma](https://www.figma.com/file/1bvSoMV4ygNqHGXmkIBgJd/Survey-workspace?type=design&node-id=8-2102&mode=design&t=lJwoAoQJBO6OR96C-0))
2. `displayMode` - value is `"map"` when `"hubMapSurvey"` is present in `typeKeywords`, `"standard"` when `hubMapSurvey` is absent in typeKeywords. (will control which tile select controls is ticked/checked, per [figma]([figma](https://www.figma.com/file/1bvSoMV4ygNqHGXmkIBgJd/Survey-workspace?type=design&node-id=8-2102&mode=design&t=lJwoAoQJBO6OR96C-0)). I'm open to suggestions for naming here)

The `questions` array from the form configuration can include nested structures to support multiple pages of questions. [hasMapQuestion](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L767) [getMapQuestion](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L776), [isMapQuestion](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L754) & [isPageQuestion](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L90) utilities exist to determine if a map question exists and will need to be ported/hoisted as part of this effort.

A [isMapSurvey](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L1005) util exists that will also need ported/hoisted to hub.js but renamed/tweaked to return the expected values enumerated in the second bullet point above.

These ported/hoisted utils can likely live in a new `packages/common/feedback/utils` dir.

I envision the code in this file looking something like:

```ts
feedback.hasMapQuestion = hasMapQuestion(model.formJSON.questions);
feedback.displayMode = getDisplayMode(model.item);
```